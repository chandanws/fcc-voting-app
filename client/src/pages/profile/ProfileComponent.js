import React, { Component } from "react";
import { Typography } from "../../Typography/Typography";
import { FormElement, Form } from "../../form/FormComponent";
import PropTypes from "prop-types";
import List, { ListItem } from "../../list/ListComponent";

export default class ProfileComponent extends Component {
  static defaultProps = {
    username: "Anonymous"
  };

  state = {
    passwordOld: "",
    passwordOne: "",
    passwordTwo: ""
  };

  handleChange = id => event => {
    if (id === 1) {
      return this.setState({ passwordOld: event.target.value });
    } else if (id === 2) {
      return this.setState({ passwordOne: event.target.value });
    } else {
      return this.setState({ passwordTwo: event.target.value });
    }
  };

  render() {
    return (
      <div>
        <Typography component="h1" text={`Hello ${this.props.username}`} />
        <hr />
        <Typography component="h2" text="Change password" />
        <Form onSubmit={e => e.preventDefault()}>
          <FormElement
            id={1}
            label="Old Password"
            type="password"
            value={this.state.value}
            onChange={this.handleChange}
          />
          <FormElement
            id={2}
            label="New Password"
            type="password"
            value={this.state.value}
            onChange={this.handleChange}
          />
          <FormElement
            id={3}
            label="New Password"
            type="password"
            value={this.state.value}
            onChange={this.handleChange}
          />
        </Form>
        <Typography component="h2" text="Your polls" />
        <List modifiers={["bordered"]}>
          <ListItem>Poll1</ListItem>
          <ListItem>Poll1</ListItem>
          <ListItem>Poll1</ListItem>
        </List>
      </div>
    );
  }
}
