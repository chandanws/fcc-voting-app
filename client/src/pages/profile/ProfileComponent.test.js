import React from "react";
import ProfileComponent from "./ProfileComponent";
import { shallow } from "enzyme";

describe("<Profilecomponent />", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<ProfileComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
