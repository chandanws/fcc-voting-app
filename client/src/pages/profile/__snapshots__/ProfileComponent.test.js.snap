// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profilecomponent /> should render correctly 1`] = `
<div>
  <Typography
    component="h1"
    text="Hello Anonymous"
  />
  <hr />
  <Typography
    component="h2"
    text="Change password"
  />
  <Form
    onSubmit={[Function]}
  >
    <Component
      component="input"
      id={1}
      label="Old Password"
      onChange={[Function]}
      type="password"
      validation=""
    />
    <Component
      component="input"
      id={2}
      label="New Password"
      onChange={[Function]}
      type="password"
      validation=""
    />
    <Component
      component="input"
      id={3}
      label="New Password"
      onChange={[Function]}
      type="password"
      validation=""
    />
  </Form>
  <Typography
    component="h2"
    text="Your polls"
  />
  <List
    modifiers={
      Array [
        "bordered",
      ]
    }
  >
    <Component
      modifiers={Array []}
    >
      Poll1
    </Component>
    <Component
      modifiers={Array []}
    >
      Poll1
    </Component>
    <Component
      modifiers={Array []}
    >
      Poll1
    </Component>
  </List>
</div>
`;
